### ISSUE TITLE
[Bug]: Android 14 always get PermissionStatus.denied after change permission status in setting

### ISSUE THREAD

**Issue by Jonty1997** [Baseflow/flutter-permission-handler#1246]
### Please check the following before submitting a new issue.

- [X] I have searched the [existing issues](https://github.com/baseflow/flutter-permission-handler/issues).
- [X] I have carefully [read the documentation](https://github.com/Baseflow/flutter-permission-handler/blob/main/permission_handler/README.md) and verified I have added the required platform specific configuration.

### Please select affected platform(s)

- [X] Android
- [ ] iOS
- [ ] Windows

### Steps to reproduce

Flutter ：3.13.9,
permission_handler : 11.0.1

mobile ：XIaoMi 13 
Android version ：14

Step：

1. In application Settings, set  permission（**photos/videos**） to deny；
2. In flutter App,user **Permission.photos.request()**, The result is always **PermissionStatus.denied**

### Expected results

should return PermissionStatus.permanentlyDenied

### Actual results

always return  **PermissionStatus.denied**

### Code sample

<details><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or video

<details>
<summary>Screenshots or video demonstration</summary>

[Upload media here]

</details>

### Version

11.0.1

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[Paste your output here]
```

</details>

**Comment by DennisKragekjaer** [comment]
Same issue on iOS. Permission.notification.request() yields PermissionStatus.denied eventhough it is granted. Started out with granted, then went to settings and turned off notifications, back to app ask again, PermissionStatus.denied. Went to setting, turn on notification, back to app, status never change.

**Comment by lzxyh0227** [comment]
me too

**Comment by TimHoogstrate** [comment]
I can reproduce this. However, I am not sure if permanently denied should be triggered because the same also happens if you use for example Location.

**Comment by mvanbeusekom** [comment]
This is correct behaviour. Android will only report a `denied` or `granted` result and not provide additional information (e.g. if permissions are permanently denied or granted only once).

This is documented in Android official documentation here: https://developer.android.com/training/permissions/requesting
